python/mlcroissant/mlcroissant/_src/operation_graph/operations/read.py

```diff
@@ -19,6 +20,50 @@
 from mlcroissant._src.structure_graph.nodes.source import FileProperty
 
 
+class ReadingMethod(enum.Enum):
+    """Reading method derived from the fields that consume the FileObject/FileSet."""
+
+    CONTENT = enum.auto()
+    JSON = enum.auto()
+    LINES = enum.auto()
+    NONE = enum.auto()
+
+
+def _reading_method(
+    node: FileObject | FileSet, fields: tuple[Field, ...]
+) -> ReadingMethod:
+    """Extracts the reading method from the fields.
+
+    If several reading methods are found, we raise an error for now. Indeed, it is
+    unlikely that the same FileObject/FileSet has to be read in different manners. Also,
+    an alternative solution is to define n FileObjects/FileSets when you have n
+    different reading methods.
+    """
+    reading_methods: set[ReadingMethod] = set()
+    for field in fields:
+        extract = field.source.extract
+        if extract.column:
+            reading_methods.add(ReadingMethod.CONTENT)
+        elif extract.file_property == FileProperty.lines:
+            reading_methods.add(ReadingMethod.LINES)
+        elif extract.file_property == FileProperty.content:
+            reading_methods.add(ReadingMethod.CONTENT)
+        elif extract.json_path:
+            reading_methods.add(ReadingMethod.JSON)
+    if len(reading_methods) == 0:
+        return ReadingMethod.NONE
+    if len(reading_methods) > 1:
+        raise ValueError(
+            f"Cannot read {node=}. The fields use several reading methods:"
+            f" {reading_methods}. Reading the same FileObject/FileSet using different"
+            " reading methods has yet to be implemented. Please, create an issue"
+            " (https://github.com/mlcommons/croissant/issues/new) if your dataset"
+            " requires this feature. Alternatively, you can use two different"
+            " FileObject/FileSet pointing to the same resource."
```

Comment:
Do we already have any croissant dataset which does so -- to cite as an example?

Reply:
No, we don't. My assumption is that there won't be such a dataset, and worst case we can still create two different FileObjects/FileSets.